src/main/java/io/fabric8/maven/docker/service/RunService.java

```diff
@@ -464,11 +467,23 @@ private void startContainer(ImageConfiguration imageConfig, String id, GavLabel
     }
 
     private void updateMappedPortsAndAddresses(String containerId, PortMapping mappedPorts) throws DockerAccessException {
-        Container container = queryService.getMandatoryContainer(containerId);
-        if (container.isRunning()) {
-            mappedPorts.updateProperties(container.getPortBindings());
-        } else {
-            log.warn("Container %s is not running anymore, can not extract dynamic ports",containerId);
+        for (int attempt = 0; attempt < UPDATE_MAPPED_PORTS_RETRIES && !Thread.currentThread().isInterrupted(); attempt++) {
+            try {
+                Container container = queryService.getMandatoryContainer(containerId);
+                if (container.isRunning()) {
+                    mappedPorts.updateProperties(container.getPortBindings());
+                } else {
+                    log.warn("Container %s is not running anymore, can not extract dynamic ports", containerId);
+                }
+                return;
+            } catch (PortBindingException e) {
+                try {
+                    log.warn("Failed to get port bindings for container %s (attempt %d of %d)", containerId, attempt, UPDATE_MAPPED_PORTS_RETRIES);
+                    Thread.sleep(UPDATE_MAPPED_PORTS_RETRY_DELAY_MS);
+                } catch (InterruptedException ex) {
+                    Thread.currentThread().interrupt();
+                }
+            }
```

Comment:
Im wondering if I should throw an exception here (possibly changing the retry loop to throw the `PortBindingException` if it was seen on the last attempt)- i.e. if it has not successfully updated the mapped port's properties I guess its best it fails loudly and indicates what has gone wrong instead of just moving on. Having said this I don't understand the process fully enough to determine what will happen if this step fails & it quietly moves on.